### DISCUSSION TITLE
cmake error

### DISCUSSION THREAD

**Discussion by rwishart1** [dglazier/clas12-elSpectro#2] [General]
HI all,

I am trying to install at the moment but I am getting a problem when I try to run cmake. I have been following the steps in the readme and setting the environment variables but after cmake ../ (or cmake3) I get the error:

CMake Error at /mnt/share/cern_CentOS7/root/v6.22.03g9/cmake/ROOTConfig.cmake:159 (message):
  ROOT component Minuit2 not found
Call Stack (most recent call first):
  CMakeLists.txt:7 (find_package)

-- Configuring incomplete, errors occurred!

Is there something else I need to set first for this to work, or any other ideas?

Cheers,
Robert

**Comment by dglazier** [comment]
> HI all,
> 
> I am trying to install at the moment but I am getting a problem when I try to run cmake. I have been following the steps in the readme and setting the environment variables but after cmake ../ (or cmake3) I get the error:
> 
> CMake Error at /mnt/share/cern_CentOS7/root/v6.22.03g9/cmake/ROOTConfig.cmake:159 (message):
> ROOT component Minuit2 not found
> Call Stack (most recent call first):
> CMakeLists.txt:7 (find_package)
> 
> -- Configuring incomplete, errors occurred!
> 
> Is there something else I need to set first for this to work, or any other ideas?
> 
> Cheers,
> Robert

Hi Robert,
  The problem is Minuit2 is not included in you ROOT build, but it is requested in cmake. I think for now you can just remove the Minuit2 in the CMakeLists file, the code wil still run wihtout it and default to Minuit.
https://github.com/dglazier/clas12-elSpectro/blob/main/CMakeLists.txt#L7
Cheers
Derek

**Reply by rwishart1** [reply]
I have removed Minuit2 from L7, rm build and started again. I still seem to get the same error as before

**Reply by dglazier** [reply]
IT is also included in the actual elSpectro CMakeLists.txt as well

https://github.com/dglazier/elSpectro/blob/6257a88640bb8480c4b23056ea4ee6ecf1e2a1f4/CMakeLists.txt#L16

So you need to remove it there too
